packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -373,7 +439,13 @@ export class BaseInvocationScope<TReturn = any> {
   }> {
     assert(this.program.account, 'Wallet is required!');
 
-    const transactionRequest = await this.fundWithRequiredCoins();
+    let transactionRequest = await this.getTransactionRequest();
+
+    if (this.addSignersCallback) {
+      transactionRequest = await this.fundWithRequiredCoins();
+    } else {
+      transactionRequest = await this.assembleTx();
+    }
```

Comment:
```suggestion
    if (this.addSignersCallback) {
      this.addSignersCallback(transactionRequest);
    }
    
    transactionRequest = await this.assembleTx();
```

Would this not work?

Reply:
No, it wouldn’t.

The first example uses the old approach for estimation and funding, while the second one uses the new method.